src/index.js

```diff
@@ -724,6 +732,23 @@ module.exports = {
         })
     );
 
+    expect.exportAssertion(
+      '<DOMElement> not to have (class|classes) <array|string>',
+      (expect, subject, value) => {
+        const expectedClasses = getAttributes(subject).class;
+
+        if (typeof value === 'string') {
+          value = getClassNamesFromAttributeValue(value);
+        }
+
+        value.forEach(namedValue =>
+          removeValueFromArray(expectedClasses, namedValue)
+        );
+
+        return expect(subject, 'to only have classes', expectedClasses);
```

Comment:
Hmm, couldn't you map it to:

```js
const classAttributeValue = getAttributes(subject).class;
const actualClasses = typeof classAttributeValue === 'string' ? getClassNamesFromAttributeValue(classAttributeValue) : [];
const expectedClasses = Array.isArray(value) ? value : [value];
expect(actualClasses, 'not to contain', ...expectedClasses);
```

That would produce a nicer error message and could also cover the `to have classes` case with a bit of flag forwarding?

Reply:
Yeah could do - IIRC I did it this way for consistency something else in U-dom, I think it was https://github.com/unexpectedjs/unexpected-dom/blob/master/src/index.js#L1329. But happy to change it round if that works better :)

Reply:
When I looked at the test I thought it was confusing that it said:

> `class="bar quux" // expected [ 'bar', 'quux' ] to equal [ 'bar' ]`

... when the user didn't actually mention `bar` in the assertion.

If the other place is equivalent to that, I think we should change that as well :hugs:

Reply:
Have pushed a commit to make the assertion run via "not to contain" :)

Reply:
Yay, thanks! :tada: